src/network/networkd-mptcp.c

```diff
@@ -0,0 +1,398 @@
+                return 1;
+        }
+
+        log_debug("MPTCP limit configured");
+
+        return 1;
+}
+
+int mp_tcp_configure_limit(Manager *m) {
+        _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL;
+        int r;
+
+        assert(m);
+        assert(m->rtnl);
+
+        if (m->mp_tcp_subflows == UINT32_MAX && m->mp_tcp_add_addr_accepted == UINT32_MAX)
+                return 0;
+
+        r = sd_genl_message_new(m->genl, SD_GENL_MPTCP, MPTCP_PM_CMD_SET_LIMITS, &req);
+        if (r < 0)
+                return log_error_errno(r, "Failed to allocate generic netlink message: %m");
+
+        if (m->mp_tcp_add_addr_accepted != UINT32_MAX) {
+                r = sd_netlink_message_append_u32(req, MPTCP_PM_ATTR_RCV_ADD_ADDRS, m->mp_tcp_add_addr_accepted);
+                if (r < 0)
+                        return log_error_errno(r, "Could not append MPTCP_PM_ATTR_RCV_ADD_ADDRS attribute: %m");
+        }
+
+        if (m->mp_tcp_subflows != UINT32_MAX) {
+                r = sd_netlink_message_append_u32(req, MPTCP_PM_ATTR_SUBFLOWS, m->mp_tcp_subflows);
+                if (r < 0)
+                        return log_error_errno(r, "Could not append MPTCP_PM_ATTR_SUBFLOWS attribute: %m");
+        }
+
+        r = netlink_call_async(m->genl, NULL, req, mp_tcp_limit_handler, NULL, m);
+        if (r < 0)
+                return log_error_errno(r, "Could not send rtnetlink message: %m");
+
+        return 0;
+}
+
+int link_configure_mp_tcp(Link *link) {
+        MPTCP *mp_tcp;
+        int r;
+
+        assert(link);
+        assert(link->network);
+
+        if (link->iftype == ARPHRD_CAN) {
+                link->mp_tcp_configured = true;
+                return 0;
+        }
+
+        link->mp_tcp_configured = false;
+        link->mp_tcp_messages = 0;
+
+        ORDERED_HASHMAP_FOREACH(mp_tcp, link->network->mp_tcp_by_section) {
+                r = mp_tcp_configure_address(link, mp_tcp);
+                if (r < 0)
+                        return r;
+        }
+
+        if (link->mp_tcp_messages == 0)
+                link->mp_tcp_configured = true;
+        else
+                log_link_debug(link, "Configuring MPTCP");
+
+        return 0;
+}
+
+static int mp_tcp_section_verify(MPTCP *mp_tcp) {
+        if (section_is_invalid(mp_tcp->section))
+                return -EINVAL;
+
+        if (mp_tcp->family == AF_UNSPEC)
+                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                         "%s: [MPTCP] section without Address= field configured. "
+                                         "Ignoring [MPTCP] section from line %u.",
+                                         mp_tcp->section->filename, mp_tcp->section->line);
+
+        return 0;
+}
+
+int network_drop_invalid_mp_tcp(Network *network) {
+        MPTCP *mp_tcp;
+
+        assert(network);
+
+        ORDERED_HASHMAP_FOREACH(mp_tcp, network->mp_tcp_by_section)
+                if (mp_tcp_section_verify(mp_tcp))
```

Comment:
this is suppsoed to drop invalid ones, hence make this clear:

```
if (mp_tcp_section_verify(…) < 0)
…
```

i.e. check explicitly for negative return, as that is usually how we encode errors, and here too